exporter/sumologicexporter/exporter.go

```diff
@@ -15,45 +15,296 @@
+				if currentMetadata != previousMetadata && previousMetadata != "" {
+					se.sendAndPushErrors(&buffer, previousMetadata, &droppedTimeSeries, &errs)
+				}
+
+				// assign metadata
+				previousMetadata = currentMetadata
+
+				// add log to the buffer
+				buffer = append(buffer, log)
+
+				// Flush buffer to avoid overlow
+				if len(buffer) == maxBufferSize {
+					se.sendAndPushErrors(&buffer, previousMetadata, &droppedTimeSeries, &errs)
+				}
+			}
+		}
+	}
+
+	// Flush pending logs
+	se.sendAndPushErrors(&buffer, previousMetadata, &droppedTimeSeries, &errs)
+
+	return droppedTimeSeries, componenterror.CombineErrors(errs)
+}
+
+// appendAndSend appends line to the body and eventually sends data to avoid exceeding the request limit
+func (se *sumologicexporter) appendAndSend(line string, pipeline string, body *strings.Builder, fields string) error {
+	var err error
+
+	if body.Len() > 0 && body.Len()+len(line) > se.config.MaxRequestBodySize {
+		err = se.send(LogsPipeline, body.String(), fields)
+		body.Reset()
+	}
+
+	if body.Len() > 0 {
+		// Do not add newline if the body is empty
+		body.WriteString("\n")
+	}
+
+	body.WriteString(line)
+	return err
+}
+
+func (se *sumologicexporter) sendLogsTextFormat(buffer []pdata.LogRecord, fields string) error {
+	body := strings.Builder{}
+	var errs []error
+
+	for j := 0; j < len(buffer); j++ {
+		err := se.appendAndSend(buffer[j].Body().StringVal(), LogsPipeline, &body, fields)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	err := se.send(LogsPipeline, body.String(), fields)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return componenterror.CombineErrors(errs)
+	}
 	return nil
 }
 
-func (se *sumologicexporter) ConsumeLogs(_ context.Context, _ pdata.Logs) error {
+func (se *sumologicexporter) sendLogsJSONFormat(buffer []pdata.LogRecord, fields string) error {
+	body := strings.Builder{}
+	var errs []error
+
+	for j := 0; j < len(buffer); j++ {
+		data := se.filterMetadata(buffer[j].Attributes(), true)
+		data[logKey] = buffer[j].Body().StringVal()
+		nextLine, err := json.Marshal(data)
+
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		err = se.appendAndSend(bytes.NewBuffer(nextLine).String(), LogsPipeline, &body, fields)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	err := se.send(LogsPipeline, body.String(), fields)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return componenterror.CombineErrors(errs)
+	}
 	return nil
 }
 
-func (se *sumologicexporter) ConsumeMetrics(_ context.Context, _ pdata.Metrics) error {
+func (se *sumologicexporter) sendLogs(buffer []pdata.LogRecord, fields string) error {
+	if se.config.LogFormat == TextFormat {
+		return se.sendLogsTextFormat(buffer, fields)
+	} else if se.config.LogFormat == JSONFormat {
+		return se.sendLogsJSONFormat(buffer, fields)
+	} else {
+		return errors.New("Unexpected log format")
+	}
```

Comment:
Seems like a good place for `switch` statement

Reply:
done